berkeley-mobile/StudyPact/CreatePreference/SelectLocationView.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  SelectLocationView.swift
+//  berkeley-mobile
+//
+//  Created by Eashan Mathur on 1/30/21.
+//  Copyright © 2021 ASUC OCTO. All rights reserved.
+//
+
+import UIKit
+
+class SelectLocationView: UIView, EnableNextDelegate {
+    let preferenceVC: CreatePreferenceViewController
+    private var buttonSelected: UIButton? {
+        didSet {
+            if let button = buttonSelected {
+                isNextEnabled = true
+                preferenceVC.preference.isVirtual = button == virtualButton
+            } else {
+                isNextEnabled = false
+            }
+        }
+    }
+    var isNextEnabled: Bool = false {
+        didSet {
+            preferenceVC.setNextEnabled()
+        }
+    }
+    
+    var virtualButton: UIButton!
+    var inPersonButton: UIButton!
+    private func createButton(title: String) -> UIButton {
+        let button = UIButton()
+        button.layer.cornerRadius = 12
+        button.layer.shadowRadius = 5
+        button.layer.shadowOpacity = 0.25
+        button.layer.shadowOffset = .zero
+        button.layer.shadowColor = UIColor.black.cgColor
+        button.layer.shadowPath = UIBezierPath(rect: button.layer.bounds.insetBy(dx: 4, dy: 4)).cgPath
+        
+        button.setTitle(title, for: .normal)
+        button.titleLabel?.font = Font.regular(16)
+        button.titleLabel?.textColor = Color.ActionButton.color
+        button.backgroundColor = .white
+        
+        button.setTitleColor(Color.blackText, for: .normal)
+        button.addTarget(self, action: #selector(toggleButton(sender:)), for: .touchUpInside)
+        
+        return button
+    }
+    
+    @objc func toggleButton(sender: ActionButton) {
+        if !sender.isSelected {
+            sender.backgroundColor = Color.ActionButton.background
+            sender.setTitleColor(.white, for: .normal)
+            sender.isSelected.toggle()
+            if let buttonSelected = self.buttonSelected {
+                buttonSelected.backgroundColor = .white
+                buttonSelected.setTitleColor(Color.blackText, for: .normal)
+                sender.backgroundColor = Color.ActionButton.background
+                sender.setTitleColor(.white, for: .normal)
+            }
+            self.buttonSelected = sender
+        }
+    }
```

Comment:
If `buttonSelected.isSelected` is never set to false when another option is chosen, so choosing "Virtual", then "In Person", then attempting to re-select "Virtual" doesn't do anything.